sox/transform.py

```diff
@@ -1875,9 +1875,10 @@ def noiseprof(self, input_filepath, profile_path):
 
         '''
         if os.path.isdir(profile_path):
-            raise ValueError("profile_path {} is a directory, but should be a file")
+            raise ValueError("profile_path {} is a directory, but filename should be specified.")
         
-        if not os.access(os.path.dirname(profile_path), os.W_OK):
+        if (bool(os.path.dirname(profile_path)) and
```

Comment:
`os.path.curdir` seems not working after `os.chdir` is called. and I found this [why-does-python-have-os-path-curdir](http://stackoverflow.com/questions/14512087/why-does-python-have-os-path-curdir)